tests/e2e/csi_snapshot_basic.go

```diff
@@ -339,29 +345,57 @@ var _ = ginkgo.Describe("Volume Snapshot Basic Test", func() {
 				err = waitForVolumeSnapshotContentToBeDeletedWithPandoraWait(ctx, snapc,
 					*volumeSnapshot.Status.BoundVolumeSnapshotContentName, pandoraSyncWaitTime)
 				gomega.Expect(err).NotTo(gomega.HaveOccurred())
+
 			}
 		}()
 
 		snapshotSize := getAggregatedSnapshotCapacityInMb(e2eVSphere, volHandle)
+		snapshotSizeStr := convertInt64ToStrMbFormat(snapshotSize)
+		framework.Logf("Aggregator snapshot size : %s", snapshotSizeStr)
 
+		//if quotaValidation is supported in VC then this block will be executed
 		if isQuotaValidationSupported && supervisorCluster {
-			totalQuotaUsedAfter, storagePolicyQuotaAfter, storagePolicyUsageAfter =
-				validateQuotaUsageAfterResourceCreation(ctx, restConfig,
-					storageclass.Name, namespace, snapshotUsage, snapshotExtensionName,
-					snapshotSize, totalQuotaUsedBefore, storagePolicyQuotaBefore,
-					storagePolicyUsageBefore)
+			//Verifies the expected quota details once the workloads are created
+			var expectedTotalStorage []string
+			expectedTotalStorage = append(expectedTotalStorage, diskSize, snapshotSizeStr)
+
+			_, quotavalidationStatus := validateTotalQuota(ctx, restConfig, storageclass.Name, namespace,
+				expectedTotalStorage, totalQuotaUsedBefore, islatebinding)
+			gomega.Expect(quotavalidationStatus).NotTo(gomega.BeFalse())
+
+			sp_quota_pvc_status, sp_usage_pvc_status := validateQuotaUsageAfterResourceCreation(ctx, restConfig,
+				storageclass.Name, namespace, pvcUsage,
+				volExtensionName, []string{diskSize}, totalQuotaUsedBefore, pvc_storagePolicyQuotaBefore,
+				pvc_storagePolicyUsageBefore, islatebinding)
+			gomega.Expect(sp_quota_pvc_status && sp_usage_pvc_status).NotTo(gomega.BeFalse())
+
+			sp_quota_Snap_status, sp_usage_Snap_status := validateQuotaUsageAfterResourceCreation(ctx, restConfig,
```

Comment:
Can we handle vm, pvc and snapshot quota in one util

If not, please give comment, else it is confusing

Reply:
validateQuotaUsageAfterResourceCreation is the only one util which does calculation for all resource 

For PVC resource -> PVCUsage extension and PVC disksize should be passed 
Similarly for Snapshot -> Snapshot extension, aggregatorSnapshotSize should be passed
For Vm service -> Vmservice extension , diskzise should be passed